src/middlewares/piwik.js

```diff
@@ -0,0 +1,40 @@
+/* global Piwik */
+import {
+  PRE_UPLOAD_FILE
+} from '../actions'
+
+const CATEGORY = {
+  INTERACTION: 'interaction'
+}
+
+const ACTIONS = {
+  UPLOAD: 'upload'
+}
+
+const piwik = store => next => action => {
+  let event = null
+
+  switch (action.type) {
+    case PRE_UPLOAD_FILE:
+      event = {
+        category: CATEGORY.INTERACTION,
+        action: ACTIONS.UPLOAD,
+        name: 'file',
+        value: action.count
+      }
+      break
+    default:
+      break
+  }
+
+  if (event && event.category && event.action) {
+    try {
+      const tracker = Piwik.getTracker()
+      tracker.trackEvent(event.category, event.action, event.name, event.value)
+    } catch (err) { }
+  }
+
+  return next(action)
+}
+
+export default piwik
```

Comment:
The file being in a folder called `middlewares` makes this pretty clear. It also seems pretty standard ([refux-logger](https://github.com/evgenyrodionov/redux-logger/blob/master/src/index.js#L126) and [redux-thunk](https://github.com/gaearon/redux-thunk/blob/master/src/index.js#L14) typically do the same). And at [import time](https://github.com/cozy/cozy-drive/blob/master/src/main.jsx#L16), the middleware-thing is kept explicit.